### PR TITLE
Add codesponsor.io badge

### PR DESCRIPTION
**Pull request by cavneb**
PR for https://github.com/badges/shields/issues/1042

screen shot 2017-09-10 at 6 12 49 pm

### CODE REVIEW

File: service-tests/codesponsor.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    name: Joi.equal('sponsor'),
+    value: Joi.string().regex(/^[0-9]+(\.[0-9]+)? BTC/)
+  }));
+
+t.create('display bitcoin amount')
+  .get('/hopsoft/bg.json')
+  .intercept(nock => nock('https://app.codesponsor.io')
+    .get('/shield/hopsoft/bg')
+    .reply(200, {
+      bitcoin: 0.004359,
```
Comment (Daniel15) [inline service-tests/codesponsor.js]: This test is going to start failing if this value ever changes... Better to do`Joi.number()` here.

Comment (cavneb) [reply service-tests/codesponsor.js]: I will fix it. Thanks you

Comment (cavneb) [reply service-tests/codesponsor.js]: @Daniel15 This is now fixed. Thanks!

**Comment by paulmelnikow** [comment]
Update: It's fine to leave the logo checked in. With #1092 it can be turned on using `?logo=codesponsor`. However, it should be turned off by default.

**Comment by paulmelnikow** [comment]
Are these requests expected to work without a valid token? I can't get them to. I've fiddled with the tests for a while.

Feel free to ping this issue or repoen. I can't test or merge this without your help.

**Comment by cavneb** [comment]
Hi @paulmelnikow I just saw this. I apologize for the delayed reply. I will make the requested changes and will provide a token to test with.

**Comment by paulmelnikow** [comment]
Sounds good, and no worries!

**Pull request closed without merging**